sql_server/pyodbc/schema.py

```diff
@@ -49,6 +49,8 @@ class DatabaseSchemaEditor(BaseDatabaseSchemaEditor):
     sql_delete_table = "DROP TABLE %(table)s"
     sql_rename_column = "EXEC sp_rename '%(table)s.%(old_column)s', %(new_column)s, 'COLUMN'"
     sql_rename_table = "EXEC sp_rename %(old_table)s, %(new_table)s"
+    sql_create_unique_null = "CREATE UNIQUE INDEX %(name)s ON %(table)s(%(columns)s) " \
+                             "WHERE %(columns)s IS NOT NULL"
```

Comment:
On testing these changes: it appears when attempting a delete of a unique nullable, the code tries to delete the constraint twice. That's because the constraint shows up as both a unique constraint and an index:

```python
{'check': False,
   'columns': [u'account_id'],
   'foreign_key': None,
   'index': True,
   'orders': ['ASC'],
   'primary_key': False,
   'type': u'idx',
   'unique': True}
```

My guess is that on like 696 we will need to add an additional check to ensure that we aren't deleting things where `infodict['unique'] == True`.

Reply:
It also appears that, in `_alter_field`, the unique nullable constraint is not being removed before a column rename, as opposed to other kinds of constraints.